File: Low Level Design (LLD)/DesignPatterns/3. Structural/Visitor/Cpp/src/NamePrinterVisitor.cpp
#include "../include/NamePrinterVisitor.h"
#include "../include/File.h"
#include "../include/Directory.h"

void NamePrinterVisitor::visit(File* file) {
    std::cout << "File: " << file->getName() << std::endl;
}

void NamePrinterVisitor::visit(Directory* directory) {
    std::cout << "Directory: " << directory->getName() << std::endl;
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Visitor/Java/NamePrinterVisitor.java
public class NamePrinterVisitor implements Visitor {
    @Override
    public void visit(File file) {
        System.out.println("File: " + file.getName());
    }

    @Override
    public void visit(Directory directory) {
        System.out.println("Directory: " + directory.getName());
    }
}
